### PR TITLE
[MBIN] Add subscribers badge

### LINKED ISSUE #10264: Mbin
### 📋 Description

Mbin subscriber count badge same as the Lemmy one.

### 🔗 Data

Public API: https://fedia.io/api/docs/

The following will get information for any named magazine:
https://{server}/api/magazine/name/{magazine name}
There's a subscriptionscount key which provides the value.

### 🎤 Motivation

Mbin exists on the Fediverse. We already have subscriber count badges for Lemmy, and Mbin will federate with it, so for consistency it would be useful to be able to get this data from Mbin too.

### PR DESCRIPTION
**Pull request by chris-y**
Add Mbin subscribers badge.  This fixes  #10264

### CODE REVIEW

File: services/mbin/mbin.service.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  static openApi = {
+    '/mbin/{magazine}': {
+      get: {
+        summary: 'Mbin',
+        description:
+          'Mbin is a fork of Kbin, a content aggregator for the Fediverse.',
+        parameters: pathParams({
+          name: 'magazine',
+          description: 'The magazine to query.  This is CASE SENSITIVE.',
```
Comment (PyvesB) [inline services/mbin/mbin.service.js:25]:
```suggestion
          description: 'The magazine to query. This is CASE SENSITIVE.',
```

File: services/mbin/mbin.tester.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+t.create('invalid magazine').get('/magazine.invalid.json').expectBadge({
+  label: 'magazine',
+  message: 'invalid magazine',
+  color: 'red',
+})
+
+t.create('test on real mbin magazine for API compliance')
+  .get('/teletext@fedia.io.json')
+  .timeout(10000)
```
Comment (PyvesB) [inline services/mbin/mbin.tester.js:58]: Is this timeout needed? I'd hope the API to respond much faster than that, many of our badges are displayed in GitHub READMEs, which have a 4s timeout for requests.

File: services/mbin/mbin.tester.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }),
+      ),
+  )
+  .expectBadge({
+    label: 'subscribe to magazine@instance.tld',
+    message: '42',
+    color: 'brightgreen',
+  })
+
+t.create('bad server or connection')
```
Comment (PyvesB) [inline services/mbin/mbin.tester.js:23]: I don't think this test adds much, it doesn't validate any special behaviour that was implemented in _mbin.service.js_, I'd suggest dropping it. :)

File: services/mbin/mbin.service.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          example: 'teletext@fedia.io',
+        }),
+      },
+    },
+  }
+
+  static defaultBadgeData = { label: 'magazine', namedLogo: 'activitypub' }
```
Comment (PyvesB) [inline services/mbin/mbin.service.js:32]: A bit of a shame that we can't use a better logo that that of the underlying protocol. But to be able to add it to simple-icons, we'd need to meet the following requirement:

> If the brand is a Fediverse project, it needs to have a minimum of 10,000 Monthly Active Users as listed on [FediDB](https://fedidb.org/software).

Mbin [is over an order of magnitude under](https://fedidb.org/software/mbin) at this point.

**Review by PyvesB — COMMENTED** [review]
Thanks for the contribution! Overall looks good to me, left a few comments and suggestions.

File: services/mbin/mbin.tester.js (hunk 3 above)
Comment (chris-y) [reply services/mbin/mbin.tester.js:23]: Removed

File: services/mbin/mbin.tester.js (hunk 2 above)
Comment (chris-y) [reply services/mbin/mbin.tester.js:58]: No not needed, responded in 208ms when I ran the test, I've removed it.

File: services/mbin/mbin.tester.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  .networkOff()
+  .expectBadge({
+    label: 'magazine',
+    message: 'inaccessible',
+    color: 'lightgrey',
+  })
+
+t.create('unknown community')
+  .get('/magazine@instance.tld.json')
+  .intercept(nock =>
```
[the remark this answers is not in the document]
Comment (chris-y) [reply services/mbin/mbin.tester.js:34]: Yes, changed to run against real server with a random ULID as the magazine name.

File: services/mbin/mbin.tester.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  message: 'invalid magazine',
+  color: 'red',
+})
+
+t.create('test on real mbin magazine for API compliance')
+  .get('/teletext@fedia.io.json')
+  .timeout(10000)
+  .expectBadge({
+    label: 'subscribe to teletext@fedia.io',
+    message: Joi.string().regex(/^[0-9]+$/),
```
[the remark this answers is not in the document]
Comment (chris-y) [reply services/mbin/mbin.tester.js]: done!

**Comment by chris-y** [comment]
Implemented all the requested changes.

**Review by PyvesB — APPROVED** [review]
Thanks for the changes, looks great!

**Pull request merged**